nomos-utils/src/lib.rs

```diff
@@ -25,7 +25,7 @@ pub mod serde {
         deserializer: D,
     ) -> Result<[u8; N], D::Error> {
         use serde::Deserialize;
-        let s = <&[u8]>::deserialize(deserializer)?;
+        let s = String::deserialize(deserializer)?;
```

Comment:
Nit: maybe `<&str>::deserialize(deserializer)` also work? If so, we can avoid allocating for some serializers implementation.

Reply:
True, I didn't find a way at first but this helped: https://github.com/serde-rs/serde/issues/1413#issuecomment-494892266